market_maker_keeper/leverjfutures_market_maker_keeper.py

```diff
@@ -0,0 +1,278 @@
+
+        parser.add_argument("--pair", type=str, required=True,
+                            help="Token pair (sell/buy) on which the keeper will operate")
+
+        parser.add_argument("--debug", dest='debug', action='store_true',
+                            help="Enable debug output")
+
+        self.arguments = parser.parse_args(args)
+    
+        if "infura" in self.arguments.rpc_host:
+            self.web3 = Web3(HTTPProvider(endpoint_uri=f"http://{self.arguments.rpc_host}",
+                                      request_kwargs={"timeout": self.arguments.rpc_timeout}))
+        else:
+            self.web3 = Web3(HTTPProvider(endpoint_uri=f"http://{self.arguments.rpc_host}:{self.arguments.rpc_port}",
+                                      request_kwargs={"timeout": self.arguments.rpc_timeout}))
+
+        self.web3.eth.defaultAccount = self.arguments.eth_from
+        register_keys(self.web3, self.arguments.eth_key)
+
+        setup_logging(self.arguments)
+
+        self.bands_config = ReloadableConfig(self.arguments.config)
+        self.price_feed = PriceFeedFactory().create_price_feed(self.arguments)
+        self.spread_feed = create_spread_feed(self.arguments)
+        self.control_feed = create_control_feed(self.arguments)
+        self.order_history_reporter = create_order_history_reporter(self.arguments)
+
+        self.history = History()
+
+        self.leverj_api = LeverjFuturesAPI(web3=self.web3,
+                                    api_server=self.arguments.leverj_api_server,
+                                    account_id=self.arguments.account_id,
+                                    api_key=self.arguments.api_key,
+                                    api_secret=self.arguments.api_secret,
+                                    timeout=self.arguments.leverj_timeout)
+
+
+        self.order_book_manager = OrderBookManager(refresh_frequency=self.arguments.refresh_frequency)
+        self.order_book_manager.get_orders_with(lambda: self.leverj_api.get_orders(self.pair()))
+        self.order_book_manager.get_balances_with(lambda: self.leverj_api.get_balances())
+        self.order_book_manager.cancel_orders_with(lambda order: self.leverj_api.cancel_order(order.order_id))
+        self.order_book_manager.enable_history_reporting(self.order_history_reporter, self.our_buy_orders,
+                                                         self.our_sell_orders)
+        self.order_book_manager.start()
+
+    def main(self):
+        with Lifecycle() as lifecycle:
+            lifecycle.initial_delay(1)
+            lifecycle.on_startup(self.startup)
+            lifecycle.every(1, self.synchronize_orders)
+            lifecycle.on_shutdown(self.shutdown)
+
+    def startup(self):
+        #quote_increment = 1/(self.leverj_api.get_product(self.arguments.pair)["ticksperpoint"])
+        quote_increment = self.leverj_api.get_product(self.pair())["tickSize"]
```

Comment:
I would actually abstract this method and put it into pyexchange...

This is what we do for other MM keepers - https://github.com/makerdao/pyexchange/blob/75f4bccf02c0b0d1e5cb4bd8b62d652a75ab93c5/pyexchange/binance_us.py#L83